server/routes/varyingLicences/handlers/policyChange.test.ts

```diff
@@ -0,0 +1,186 @@
+      await handler.GET(req, res)
+      expect(res.redirect).toHaveBeenCalledWith('/licence/vary/id/1/policy-changes')
+    })
+
+    describe('page rendering', () => {
+      it('should redirect to the deleted condition page when the condition change type is DELETED', async () => {
+        const condition = { code: 'code5', text: 'Conditon 5', category: 'group1', requiresInput: false }
+        conditionService.getAdditionalConditionByCode.mockResolvedValue(condition)
+        await handler.GET(req, res)
+
+        expect(res.render).toHaveBeenCalledWith('pages/vary/policyConditionDeleted', {
+          condition: condition1,
+          conditionCounter: 1,
+          conditionHintText: undefined,
+          licenceId: '1',
+          policyChangesCount: 4,
+          replacements: [condition],
+        })
+      })
+
+      it('should redirect to the text change page when the condition change type is TEXT_CHANGE', async () => {
+        req.params.changeCounter = '2'
+        await handler.GET(req, res)
+
+        expect(res.render).toHaveBeenCalledWith('pages/vary/policyTextChange', {
+          condition: condition2,
+          conditionCounter: 2,
+          conditionHintText: undefined,
+          licenceId: '1',
+          policyChangesCount: 4,
+        })
+      })
+
+      it('should redirect to the new options page when the condition change type is NEW_OPTIONS', async () => {
+        req.params.changeCounter = '3'
+        await handler.GET(req, res)
+
+        expect(res.render).toHaveBeenCalledWith('pages/vary/policyNewOptions', {
+          condition: condition3,
+          conditionCounter: 3,
+          conditionHintText: undefined,
+          licenceId: '1',
+          policyChangesCount: 4,
+        })
+      })
+
+      it('should redirect to the replaced condition page when the condition change type is REPLACED', async () => {
+        const condition = { code: 'code6', text: 'Conditon 6', category: 'group1', requiresInput: false }
+        conditionService.getAdditionalConditionByCode.mockResolvedValue(condition)
+        req.params.changeCounter = '4'
+        await handler.GET(req, res)
+
+        expect(res.render).toHaveBeenCalledWith('pages/vary/policyConditionReplaced', {
+          condition: condition4,
+          conditionCounter: 4,
+          conditionHintText: undefined,
+          licenceId: '1',
+          policyChangesCount: 4,
+          replacements: [condition],
+        })
+      })
+    })
+  })
+
+  describe('POST', () => {
+    it('should redirect to the policyChangesInputCallback', async () => {
+      await handler.POST(req, res)
+      expect(res.redirect).toHaveBeenCalledWith('/licence/vary/id/1/policy-changes/input/callback/1')
+    })
+
+    it('sets the changeConditionInputs in session storage', async () => {
+      req.body.additionalConditions = ['code 5', 'code 7']
+      condition1 = { ...condition1, suggestions: [{ code: 'code 5' }, { code: 'code 7' }] } as LicenceConditionChange
```

Comment:
I would assign a new variable here, rather than overwrite the top level variable. It's fine here, but if other tests are added at a later point, they may rely on the initial state of the object.

Reply:
Copying and pasting response to the other related comment for completion's sake

> Apologies, I did look into this but I later pass in conditions1-4 as mocked responses/ session objects, which I would need to redefine if I were to create a new variable in the individual tests. It can definitely be done, but I did it this way to avoid repetition of code.